testing/regress/ecl/workflow_10.ecl

```diff
@@ -0,0 +1,48 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2020 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the 'License');
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an 'AS IS' BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+//version parallel=false
+//version parallel=true
+
+import ^ as root;
+optParallel := #IFDEFINED(root.parallel, false);
+
+#option ('parallelWorkflow', optParallel);
+#option('numWorkflowThreads', 5);
+
+display(Integer8 thisInteger) := FUNCTION
+  ds := dataset([thisInteger], {Integer8 value});
+  RETURN Output(ds, NAMED('logging'), EXTEND);
+END;
+
+names := RECORD
+  String8 firstname;
+  String8 surname;
+END;
+
+
+dset := DATASET([{'nathan', 'halliday'}, {'margaret', 'thatcher'}], names);
+
+//A := OUTPUT(ds,,'myfile', OVERwrite);
+//MyFile := DATASET('myfile', names, thor);
+
+X := display(5) : independent;
+Y := display(10) : independent;
+C := COUNT(dset) < 4 : independent;
```

Comment:
Use COUNT(NOFOLD(dset)) to ensure that the condition isn't evaluated at compile time.